src/main/java/io/gravitee/repository/management/model/Event.java

```diff
@@ -16,113 +16,117 @@
-	 */
-	private EventType type;
-
-	/**
-	 * The event payload
-	 */
-	private String payload;
-
-	/**
-	 * The event parent
-	 */
-	private String parentId;
-	
-	/**
-	 * The event origin
-	 */
-	private String origin;
-	
-	/**
-	 * The event publisher
-	 */
-	private String username;
-
-	/**
-	 * The event creation date
-	 */
-	private Date createdAt;
-
-	/**
-	 * The event last updated date
-	 */
-	private Date updatedAt;
-
-	public String getId() {
-		return id;
-	}
-
-	public void setId(String id) {
-		this.id = id;
-	}
-
-	public EventType getType() {
-		return type;
-	}
-
-	public void setType(EventType type) {
-		this.type = type;
-	}
-
-	public String getPayload() {
-		return payload;
-	}
-
-	public void setPayload(String payload) {
-		this.payload = payload;
-	}
-
-	public String getParentId() {
-		return parentId;
-	}
-
-	public void setParentId(String parentId) {
-		this.parentId = parentId;
-	}
-
-	public String getOrigin() {
-		return origin;
-	}
-
-	public void setOrigin(String origin) {
-		this.origin = origin;
-	}
-	
-	public String getUsername() {
-		return username;
-	}
-
-	public void setUsername(String username) {
-		this.username = username;
-	}
-
-	public Date getCreatedAt() {
-		return createdAt;
-	}
-
-	public void setCreatedAt(Date createdAt) {
-		this.createdAt = createdAt;
-	}
-
-	public Date getUpdatedAt() {
-		return updatedAt;
-	}
-
-	public void setUpdatedAt(Date updatedAt) {
-		this.updatedAt = updatedAt;
-	}
+    /**
+     * The event ID
+     */
+    private String id;
+
+    /**
+     * The event Type
+     */
+    private EventType type;
+
+    /**
+     * The event payload
+     */
+    private String payload;
+
+    /**
+     * The event parent
+     */
+    private String parentId;
+
+    /**
+     * The event properties
+     */
+    private Map<String, String> properties;
+
+    /**
+     * The event creation date
+     */
+    private Date createdAt;
+
+    /**
+     * The event last updated date
+     */
+    private Date updatedAt;
+
+    public String getId() {
+        return id;
+    }
+
+    public void setId(String id) {
+        this.id = id;
+    }
+
+    public EventType getType() {
+        return type;
+    }
+
+    public void setType(EventType type) {
+        this.type = type;
+    }
+
+    public String getPayload() {
+        return payload;
+    }
+
+    public void setPayload(String payload) {
+        this.payload = payload;
+    }
+
+    public String getParentId() {
+        return parentId;
+    }
+
+    public void setParentId(String parentId) {
+        this.parentId = parentId;
+    }
+
+    public Map<String, String> getProperties() {
+        return properties;
+    }
+
+    public void setProperties(Map<String, String> properties) {
+        this.properties = properties;
+    }
+
+    public Date getCreatedAt() {
+        return createdAt;
+    }
+
+    public void setCreatedAt(Date createdAt) {
+        this.createdAt = createdAt;
+    }
+
+    public Date getUpdatedAt() {
```

Comment:
Do we really need an updated date for an event ?
According to me, an event must be immuable.